website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,764 @@
+        schema_name: metaSchema.name,
+        schema_version: metaSchema.version,
+        schema_data: schemaData
+    }).then(self.updateData.bind(self));
+};
+RegistrationEditor.prototype.putSaveData = function(payload) {
+    var self = this;
+    $osf.putJSON(self.urls.update.replace('{draft_pk}', self.draft().pk), payload).then(self.updateData.bind(self));
+};
+/**
+ * Save the current Draft
+ **/
+RegistrationEditor.prototype.save = function() {
+    var self = this;
+    var metaSchema = self.draft().metaSchema;
+    var schema = metaSchema.schema;
+    var data = {};
+    $.each(schema.pages, function(i, page) {
+        $.each(page.questions, function(qid, question) {
+            if (question.type === 'object') {
+                var value = {};
+                $.each(question.properties, function(prop, subQuestion) {
+                    value[prop] = {
+                        value: subQuestion.value(),
+                    };
+                });
+                data[qid] = value;
+            } else {
+                data[qid] = {
+                    value: question.value(),
+                };
+            }
+        });
+    });
+
+    if (typeof self.draft().pk === 'undefined') {
+        self.create(self);
+    } else {
+        self.putSaveData({
+            schema_name: metaSchema.name,
+            schema_version: metaSchema.version,
+            schema_data: data
+        });
+    }
+    return true;
+};
+
+/**
+ * @class RegistrationManager
+ * Model for listing DraftRegistrations
+ *
+ * @param {Object} node: optional data to instatiate model with
+ * @param {String} draftsSelector: DOM node to bind VM to
+ * @param {Object} urls:
+ * @param {String} urls.list:
+ * @param {String} urls.get:
+ * @param {String} urls.delete:
+ * @param {String} urls.edit:
+ * @param {String} urls.schemas:
+ **/
+var RegistrationManager = function(node, draftsSelector, urls) {
+    var self = this;
+
+    self.node = node;
+    self.draftsSelector = draftsSelector;
+
+    self.urls = urls;
+
+    self.schemas = ko.observableArray();
+    self.selectedSchema = ko.observable({
+        description: ''
+    });
+
+    // TODO: convert existing registration UI to frontend impl.
+    // self.registrations = ko.observable([]);
+    self.drafts = ko.observableArray();
+
+    self.loading = ko.observable(true);
+
+    self.preview = ko.observable(false);
+
+    // bound functions
+    self.getDraftRegistrations = $.getJSON.bind(null, self.urls.list);
+    self.getSchemas = $.getJSON.bind(null, self.urls.schemas);
+
+    self.previewSchema = ko.computed(function() {
+        var schema = self.selectedSchema();
+        return {
+            schema: schema.schema,
+            readonly: true
+        };
+    });
+};
+RegistrationManager.prototype.init = function() {
+    var self = this;
+
+    $osf.applyBindings(self, self.draftsSelector);
+
+    var getSchemas = self.getSchemas();
+
+    getSchemas.then(function(response) {
```

Comment:
Should there also be an error handling callback for each of the requests sent in this method!